### DISCUSSION TITLE
Is Next.js good for building web application that requires user sign-in and fetches data from an HTTP API backend?

### DISCUSSION THREAD

**Discussion by ksze** [vercel/next.js#15332] [Help]
I'm pretty sure that Next.js is good for web sites/apps where most of the content is either static or completely derived from user input, requiring no HTTP request to an HTTP API backend.

But is Next.js good for web apps that require user sign-in and that displays data that all come from requests to an HTTP API backend?

**Comment by arunoda** [comment]
Why not? It's totally possible.
[Vercel dashboard](https://vercel.com/dashboard) is a great example of this.

For auth, you can try using [next-auth](https://next-auth.js.org/) or Auth0

**Comment by THPubs** [comment]
@arunoda Does the Vercel dashboard use SSR or simply skip it using dynamic imports since the layout need to be loaded based on the logged in user (Normally we keep the user session in the browser.)?
